src/client.rs

```diff
@@ -538,10 +608,19 @@ where
             )
             .await?;
 
-        if let Some(updated_timestamp) = self.tuf.update_timestamp(&raw_signed_timestamp)? {
-            let latest_version = MetadataVersion::Number(updated_timestamp.version());
-            self.store_metadata(&timestamp_path, &latest_version, &raw_signed_timestamp)
-                .await;
+        if self.tuf.update_timestamp(&raw_signed_timestamp)?.is_some() {
+            /////////////////////////////////////////
+            // TUF-1.0.9 §5.2.4:
+            //
+            //     Persist timestamp metadata. The client MUST write the file to non-volatile
+            //     storage as FILENAME.EXT (e.g. timestamp.json).
+
+            self.store_metadata(
+                &timestamp_path,
+                &MetadataVersion::None,
```

Comment:
Can you explain why this change is safe? It seems like if we were using local metadata at all this change would break that logic.